### PR TITLE
Add permanent deletion support

### PR DESCRIPTION
**Pull request by thewheat**
Add support for permanent deletion https://developers.intercom.com/reference#delete-users
Deprecate `delete` and replace it with `archive`

### CODE REVIEW

File: intercom-java/src/test/resources/permanent_delete_response.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "id": 123456
+}
```
Comment (eripe970) [inline intercom-java/src/test/resources/permanent_delete_response.json]: Missing newline.

File: intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteRequest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+@SuppressWarnings("UnusedDeclaration")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class UserPermanentDeleteRequest {
+    @JsonProperty("intercom_user_id")
+    private String intercom_user_id;
+    public UserPermanentDeleteRequest() {
+    }
+    public UserPermanentDeleteRequest(String intercom_user_id) {
```
Comment (eripe970) [inline intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteRequest.java]: Better naming to call `intercom_user_id` => `intercomUserId`?

File: intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteRequest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@SuppressWarnings("UnusedDeclaration")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class UserPermanentDeleteRequest {
```
Comment (eripe970) [inline intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteRequest.java]: I think this should be renamed to `PermanentlyDeleteUserRequest`.

File: intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteResponse.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package io.intercom.api;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@SuppressWarnings("UnusedDeclaration")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class UserPermanentDeleteResponse {
```
Comment (eripe970) [inline intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteResponse.java]: I think this should be renamed to `PermanentlyDeleteUserResponse`.

File: intercom-java/src/main/java/io/intercom/api/User.java [hunk 5]
```diff
@@ -55,13 +55,40 @@ public static User update(User user) throws InvalidException, AuthorizationExcep
         return DataResource.update(UserUpdate.buildFrom(user), "users", User.class);
     }
 
-    public static User delete(String id)
+    /**
+     * @deprecated  Replaced by {@link #archive(String)}. Renamed for consistency with API language
+     */
+    public static User delete(String id){
```
Comment (eripe970) [inline intercom-java/src/main/java/io/intercom/api/User.java]: Minor, formatting.

File: intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteRequest.java (hunk 2 above)
Comment (thewheat) [reply intercom-java/src/main/java/io/intercom/api/UserPermanentDeleteRequest.java]: Ah yes thanks for catching that 👍

File: intercom-java/src/main/java/io/intercom/api/User.java (hunk 5 above)
Comment (thewheat) [reply intercom-java/src/main/java/io/intercom/api/User.java]: Thanks for the review @eripe970. Not sure I get exactly which formatting that you're referring to here so would be great if I could get a bit more details.

Will change some the issues that I've responded to but will leave the naming convention to the other dev teammates to take a look at 😄

Comment (eripe970) [reply intercom-java/src/main/java/io/intercom/api/User.java]: Cool, normally it's a space before the {, like `delete(String id){` => `delete(String id) {`. At least I think that's the convention that is used in a lot of code-bases.

Comment (thewheat) [reply intercom-java/src/main/java/io/intercom/api/User.java]: Ah got it 👍  I do see the other files do have that space so added it in

**Comment by eripe970** [comment]
Is it possible to do a new release that includes this PR @choran?

**Comment by thewheat** [comment]
@eripe970 looks like version 2.4.0 does contain the code in question 

image


Also available in bintray (Updated on 8th June): https://bintray.com/intercom/intercom-maven/intercom-java/2.4.0

image


Updated my dependency in my local code and I am able to make a permanent deletion request 😄 

image

**Comment by eripe970** [comment]
Thanks! 👍 Missed that it was a new release out!

**Pull request merged**